### PR TITLE
bt: rework heap usage and improvements

### PR DESCRIPTION
**Pull request by sylvioalves**

### CODE REVIEW

File: zephyr/esp32c3/src/bt/esp_bt_adapter.c [hunk 1]
```diff
@@ -26,6 +26,9 @@
 #include "esp32c3/rom/ets_sys.h"
 #include <rom/efuse.h>
 #include <riscv/interrupt.h>
+#include "esp_timer.h"
+// #include "esp_sleep.h"
```
Comment (marekmatej) [inline zephyr/esp32c3/src/bt/esp_bt_adapter.c:30]: it seems this is not used here

File: zephyr/esp32c3/CMakeLists.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 if(CONFIG_SOC_ESP32C3)
 
-  zephyr_compile_options(-Wno-unused-variable -Wno-maybe-uninitialized)
+  zephyr_compile_options(-Wno-unused-variable -Wno-unused-function -Wno-maybe-uninitialized)
```
Comment (marekmatej) [inline zephyr/esp32c3/CMakeLists.txt:5]: couldn't this be solved by configuration?

File: zephyr/esp32c3/src/bt/esp_bt_adapter.c [hunk 3]
```diff
@@ -794,18 +861,22 @@ static bool async_wakeup_request(int event)
...
-		request_lock = true;
-		break;
-	default:
-		return;
+		case BTDM_ASYNC_WAKEUP_SRC_VHCI:
+		case BTDM_ASYNC_WAKEUP_SRC_DISA:
+			allow_to_sleep = true;
+			break;
+		default:
+			allow_to_sleep = true;
```
Comment (LucasTambor) [inline zephyr/esp32c3/src/bt/esp_bt_adapter.c:875]: Can't see the need for the switch statement here, should be allow_to_sleep = false; on default?

Comment (sylvioalves) [reply zephyr/esp32c3/src/bt/esp_bt_adapter.c:875]: Same reply as in S3 BT support. I kept original file functions as is. Suggest not to change that for now.

**Comment by sylvioalves** [comment]
@LucasTambor @marekmatej PTAL again.

**Review by LucasTambor — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**